### PR TITLE
YD-344 Add buddy link to goal conflict message, if from buddy

### PR DESCRIPTION
**Pull request by Bert-R**
Along with this, refactored the use of SenderInfo to deduplicate and streamline the code

### CODE REVIEW

File: core/src/main/java/nu/yona/server/messaging/entities/Message.java [hunk 1]
```diff
@@ -90,9 +91,9 @@ public void setRead(boolean isRead)
 	 * @return The ID of the related anonymized user. Might be null if that user was already deleted at the time this message was
 	 *         sent on behalf of that user.
 	 */
-	public UUID getRelatedUserAnonymizedID()
+	public Optional<UUID> getRelatedUserAnonymizedID()
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/entities/Message.java]: We may in a subsequent PR rename this to `getSenderUserAnonymizedID()`.

File: core/src/main/java/nu/yona/server/messaging/service/BuddyMessageDTO.java [hunk 2]
```diff
@@ -7,41 +7,46 @@
...
-	public String getMessage()
+	@Component
+	public static abstract class Manager extends MessageDTO.Manager
 	{
-		return message;
+		@Override
+		protected SenderInfo getSenderInfoExtensionPoint(Message messageEntity)
+		{
+			// The buddy entity does not contain the user anonymized ID yet
+			BuddyMessage responseMmessageEntity = (BuddyMessage) messageEntity;
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/service/BuddyMessageDTO.java]: Correct spelling of `responseMmessageEntity`

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyDTO.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ private String determineTempNickname(Translator translator)
 
 	public static BuddyDTO createInstance(Buddy buddyEntity)
 	{
-		return new BuddyDTO(buddyEntity.getID(), UserDTO.createInstanceIfNotNull(buddyEntity.getUser()),
+		return new BuddyDTO(buddyEntity.getID(), UserDTO.createInstance(buddyEntity.getUser()),
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyDTO.java]: Why did you change this line? It may be that the buddy user is removed, isn't it?

Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyDTO.java]: Ah, I see why now.

**Comment by aristotelos** [comment]
Looks great, good code improvements. I detected a spelling error but can merge it for the rest.

File: core/src/main/java/nu/yona/server/messaging/entities/Message.java (hunk 1 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/entities/Message.java]: Agreed

File: core/src/main/java/nu/yona/server/messaging/service/BuddyMessageDTO.java (hunk 2 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/service/BuddyMessageDTO.java]: Beyond this: it's not a response message but a buddy message. Corrected it.

**Comment by Bert-R** [comment]
Pushed an update for the variable name.
